pkg/process/procutil/process_linux_test.go

```diff
@@ -751,7 +752,30 @@ func TestBootTimeLocalFS(t *testing.T) {
 	defer probe.Close()
 	expectT, err := host.BootTime()
 	assert.NoError(t, err)
-	assert.Equal(t, expectT, probe.bootTime)
+	assert.Equal(t, expectT, atomic.LoadUint64(&probe.bootTime))
+}
+
+func TestBootTimeRefresh(t *testing.T) {
+	os.Setenv("HOST_PROC", "resources/test_procfs/proc/")
+	bootTimeRefreshInterval = 500 * time.Millisecond
+	probe := getProbeWithPermission()
+	defer probe.Close()
+
+	assert.Equal(t, uint64(1606127264), atomic.LoadUint64(&probe.bootTime))
+	err := os.Rename("resources/test_procfs/proc/stat", "resources/test_procfs/proc/stat_temp")
+	require.NoError(t, err)
+	err = os.Rename("resources/test_procfs/proc/stat2", "resources/test_procfs/proc/stat")
+	require.NoError(t, err)
+
+	// wait for 1s so the bootTime is refreshed.
+	// This is not deterministic, but I think it's good enough to test the behavior
+	time.Sleep(time.Second)
```

Comment:
Why not just allow for passing of a custom tick function so you can deterministically control the behavior of the function? 
From process-client: 
https://github.com/DataDog/dd-go/pull/35046/files#diff-decff8aa76091a28df16ea7f11b024c9268f207d3f3bae7fe34b3f1d04a126e8

Reply:
Sorry I mean it is deterministic in the sense that we use `Sleep` with a longer period than the ticker's tick interval to make sure it's indeed updated, maybe it should use "less ideal" to describe it in that we are relying on waiting on tickers to test the changes, instead of deterministically calling functions to update attributes.

Reply:
What about using `assert.Eventually` instead of `time.Sleep`?